### PR TITLE
Master mass produce arm

### PR DESCRIPTION
**Pull request by amoyaux**
Enterprise PR: https://github.com/odoo/enterprise/pull/53901
Upgrade PR: https://github.com/odoo/upgrade/pull/5508

### CODE REVIEW

**Review by svs-odoo — APPROVED** [review]
About the fields `component_separator`, `lots_separator` and `lots_quantity_separator`, don't you think the should be included in the `production_text_help` in case at least one component is tracked by lots ?
They seems a bit cryptic and the fields itself are displayed only when debug is enable.

Also, if somebody really want to use them but with other characters than the default ones, they will have to change them every time. Maybe they should be system parameters instead of fields ?
I'm not sure, it could be overkill and can done afterward anyway, or those fields maybe should be removed and let that "static"

File: addons/mrp/wizard/mrp_batch_produce.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class MrpBatchProduct(models.TransientModel):
+    _name = 'mrp.batch.produce'
+    _description = 'Produce a batch of production order'
+
+    production_id = fields.Many2one('mrp.production', 'Production')
+    production_ids = fields.Many2many('mrp.production', string='Productions to validate')
```
[the remark this answers is not in the document]
Comment (amoyaux) [reply addons/mrp/wizard/mrp_batch_produce.py:17]: oups forget to use it. I added it back before button_mark_done in _production_text_to_object

File: addons/mrp/wizard/mrp_batch_produce.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+            if wizard.lot_name:
+                continue
+            wizard.lot_name = self.env['stock.lot']._get_next_serial(self.production_id.company_id, self.production_id.product_id)
+
+    @api.depends('production_qty')
+    def _compute_lot_qty(self):
+        for wizard in self:
+            wizard.lot_qty = wizard.production_qty
+
+    @api.depends('production_id', 'lot_name', 'component_separator')
```
[the remark this answers is not in the document]
Comment (amoyaux) [reply addons/mrp/wizard/mrp_batch_produce.py:43]: leftover from old change. Useless now

File: addons/mrp/wizard/mrp_batch_produce.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        ml_vals['lot_id'] = lots[(lot_name, move.product_id)].id
+        ml_vals['picked'] = True
+        return ml_vals
+
+    def _get_lot_and_qty(self, move, text):
+        if self.lots_quantity_separator in text:
+            lot_name, qty = re.match(r'(.+)%s(.+)' % (self.lots_quantity_separator), text).groups()
+            return lot_name, float(qty)
+        elif move.product_id.tracking == "none":
+            return False, float(text)
```
[the remark this answers is not in the document]
Comment (amoyaux) [reply addons/mrp/wizard/mrp_batch_produce.py:160]: Not really used but for safety and it makes sense to check it.

File: addons/mrp/wizard/mrp_batch_produce.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class MrpBatchProduct(models.TransientModel):
+    _name = 'mrp.batch.produce'
+    _description = 'Produce a batch of production order'
+
+    production_id = fields.Many2one('mrp.production', 'Production')
+    produced_qty = fields.Float('Produced Quantity', digits='Product Unit of Measure')
```
Comment (svs-odoo) [inline addons/mrp/wizard/mrp_batch_produce.py:17]: I think this field should be removed too (unused)

**Comment by amoyaux** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**